code/game/objects/items/weapons/mop.dm

```diff
@@ -33,9 +40,31 @@
 			if(T)
 				T.clean(src, user)
 			user << SPAN_NOTICE("You have finished mopping!")
+	else
+		makeWet(A, user)
+
+
+/obj/item/weapon/mop/proc/makeWet(atom/A, mob/user)
+	for(var/container in get_wet_containers)
+		if(istype(A, container))
+			if(A.reagents)
+				if(!A.is_open_container())
```

Comment:
oh i see you already used is_open_container.
That should be fine then, i'd say just remove the whitelist. unless there's a specific reason you feel its necessary?

Reply:
Well, this is necessary. If we make it only with is_open_container, that means mop will getting wet by unexpected items, such vials or bottles. We can add w_class check, but this one not saves us from issues in future. So, whitelist is simple and reliable. Or maybe i miss something.

Reply:
not much wrong with those, as long as they're open. You can simply rationalise it as pouring the thing onto the mop

Reply:
Okay, whitelist removed